drivers/serial/uart_gecko.c

```diff
@@ -0,0 +1,421 @@
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+	uint32_t flags = USART_IntGet(config->base);
+
+	return (flags & USART_IF_TXBL) != 0;
+}
+
+static int uart_gecko_irq_tx_ready(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+	uint32_t mask = USART_IEN_TXBL;
+
+	return (config->base->IEN & mask)
+		&& uart_gecko_irq_tx_complete(dev);
+}
+
+static void uart_gecko_irq_rx_enable(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+	uint32_t mask = USART_IEN_RXDATAV;
+
+	USART_IntEnable(config->base, mask);
+}
+
+static void uart_gecko_irq_rx_disable(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+	uint32_t mask = USART_IEN_RXDATAV;
+
+	USART_IntDisable(config->base, mask);
+}
+
+static int uart_gecko_irq_rx_full(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+	uint32_t flags = USART_IntGet(config->base);
+
+	return (flags & USART_IF_RXDATAV) != 0;
+}
+
+static int uart_gecko_irq_rx_ready(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+	uint32_t mask = USART_IEN_RXDATAV;
+
+	return (config->base->IEN & mask)
+		&& uart_gecko_irq_rx_full(dev);
+}
+
+static void uart_gecko_irq_err_enable(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+
+	USART_IntEnable(config->base, USART_IF_RXOF |
+			 USART_IF_PERR |
+			 USART_IF_FERR);
+}
+
+static void uart_gecko_irq_err_disable(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+
+	USART_IntDisable(config->base, USART_IF_RXOF |
+			 USART_IF_PERR |
+			 USART_IF_FERR);
+}
+
+static int uart_gecko_irq_is_pending(struct device *dev)
+{
+	return uart_gecko_irq_tx_ready(dev) || uart_gecko_irq_rx_ready(dev);
+}
+
+static int uart_gecko_irq_update(struct device *dev)
+{
+	return 1;
+}
+
+static void uart_gecko_irq_callback_set(struct device *dev,
+				       uart_irq_callback_t cb)
+{
+	struct uart_gecko_data *data = dev->driver_data;
+
+	data->callback = cb;
+}
+
+static void uart_gecko_isr(void *arg)
+{
+	struct device *dev = arg;
+	struct uart_gecko_data *data = dev->driver_data;
+
+	if (data->callback) {
+		data->callback(dev);
+	}
+}
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+
+static void uart_gecko_init_pins(struct device *dev)
+{
+	const struct uart_gecko_config *config = dev->config->config_info;
+
+	GPIO_Port_TypeDef tx_port;
+	unsigned int tx_pin;
+	GPIO_Mode_TypeDef tx_mode = gpioModePushPull;
+	unsigned int tx_out = 1;
+
+	GPIO_Port_TypeDef rx_port;
+	unsigned int rx_pin;
+	GPIO_Mode_TypeDef rx_mode = gpioModeInput;
+	unsigned int rx_out = 1;
+
+	unsigned int route = USART_ROUTE_RXPEN | USART_ROUTE_TXPEN;
+
+	/* TODO: This location to pin mapping is device and package specific,
+	 * move somehow to soc
+	 */
+	if (config->base == UART0) {
+#ifdef CONFIG_UART_GECKO_0
```

Comment:
This code that handles the uart location is a dirty workaround. I would like to do it with the device tree.
Could i add a location property to the uart and usart yaml for the efm32wg to remove the configuration option CONFIG_UART_GECKO_0_GPIO_LOC from Kconfig?
The used GPIOs, e.g. port F pin 6 and 7 for location 0 depend on the mcu model and package. What would be the appropriate location for this table (location -> pin), and how should this be done?

Reply:
I wonder if we should move this into the board code for now.

Reply:
I moved this into the soc folder, similar to the atmel sam soc.